src/java/com/zimbra/cs/nginx/NginxLookupExtension.java

```diff
@@ -889,6 +899,11 @@ private Server lookupUpstreamImapServer(NginxLookupRequest req) throws ServiceEx
 
                     if (imapServerAddrs == null || imapServerAddrs.length == 0) {
                         imapServerAddrs = new String[]{server.getServiceHostname()};
+                        servers.put(server.getServiceHostname(), false);
+                    }
```

Comment:
nit: remove empty line and put "else" on the same line as "}"

Reply:
done